clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -16,7 +16,7 @@
     private HubProtocol protocol;
     private Gson gson = new Gson();
 
-    public Boolean connected = false;
+    public HubConnectionState connectionState = HubConnectionState.DISCONNECTED;
```

Comment:
AFAIK, in Java it is more common to expose public members through getters and setters, like `getConnectionState()` and `setConnectionState()`

Reply:
Also, this shouldn't be public, we don't expose connection state in any of the other clients

Reply:
> it is more common to expose public members through getters and setters, like getConnectionState() and setConnectionState()
Yup, I'll change it to use a getter/setter

Reply:
> we don't expose connection state in any of the other clients

We do in the C# client
https://github.com/aspnet/SignalR/blob/2d4fb0af6fd2ef2e3a059a2a15a56484d8800d35/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs#L82-L98

Reply:
Oh, that's new. Since 2.2 I guess we do...